sparql-helpers/token-claim.sparql.js

```diff
@@ -0,0 +1,51 @@
+import { sparqlEscapeUri, sparqlEscapeDateTime, query } from 'mu';
+import { updateSudo } from '@lblod/mu-auth-sudo';
+import { PREFIXES } from '../constants';
+
+export async function createTokenClaim(tokenClaimURI, createdDate, userURI, collaborationActivityURI, graph) {
+  // Create ext:TokenClaim
+  return await updateSudo(`
+    ${PREFIXES}
+    INSERT DATA {
+        GRAPH ${sparqlEscapeUri(graph)} {
+            ${sparqlEscapeUri(tokenClaimURI)}               a                       ext:TokenClaim;
+                                                            dct:created             ${sparqlEscapeDateTime(createdDate)};
+                                                            prov:wasAttributedTo    ${sparqlEscapeUri(userURI)}.
+            ${sparqlEscapeUri(collaborationActivityURI)}    prov:generated          ${sparqlEscapeUri(tokenClaimURI)}.
+        }
+    }
+    `);
+}
+
+export async function deleteTokenClaim(tokenClaimURI, collaborationActivityURI, graph) {
+  // Delete ext:TokenClaim
+  return await updateSudo(`
+    ${PREFIXES}
+    DELETE {
+      GRAPH ${sparqlEscapeUri(graph)} {
+          ${sparqlEscapeUri(tokenClaimURI)}               a                       ext:TokenClaim;
+                                                          dct:created             ?createdDate;
+                                                          prov:wasAttributedTo    ?userUri.
+          ${sparqlEscapeUri(collaborationActivityURI)}    prov:generated          ${sparqlEscapeUri(tokenClaimURI)}.
+      }
+    } WHERE {
+      GRAPH ${sparqlEscapeUri(graph)} {
+            ${sparqlEscapeUri(tokenClaimURI)}               a                       ext:TokenClaim;
+                                                            dct:created             ?createdDate;
+                                                            prov:wasAttributedTo    ?userUri.
```

Comment:
Bij delete is het niet nodig om alle attributes op te lijsten. We willen alle info die we hebben van de token-claim verwijderen (`<token-claim> ?p ?o`).